### PR TITLE
[Docs] Edit build and install docs for upcoming 1.4 release

### PR DESCRIPTION
**Pull request by woju**
I'd prefer this PR be merged after #1102, because although it's unrelated, I think the discussion if we support RHEL 9 et.al. should be concluded before we agree on the exact language about installing on RHEL and derivatives. Regardless of that section's intro text, the CLI commands will be the same.

## How to test this PR? 

Just before release there will be packages available in `unstable-` repositories. I'll change this text when this happens. Then you should go though all the new install instructions, but for our repos prefix the names with `unstable-` (like, instead of `https://packages.gramineproject.io jammy`, write `https://packages.gramineproject.io unstable-jammy`) and see if they work. Obviously the packages from stable repos won't be available until after the release.

- - -
This change is [Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1136)

### CODE REVIEW

**Review by dimakuv — CHANGES_REQUESTED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1136)* status: all files reviewed, 4 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @woju)

*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/1136#-NMDo8r09St3XvBsnnbS:-NMDo8r12O3PiZKHVRzH:b-xl691j) (no related file):*
Where do we mention that `gramine-oot` and `gramine-dcap` are removed from the list of packages? Only in Release Notes? Not somewhere in the Docs?

___
___
*[`Documentation/quickstart.rst` line 15 at r1](https://reviewable.io/reviews/gramineproject/gramine/1136#-NMDn6261K97h-nKw_xl:-NMDn6277JJFOZkjHFB7:bb82a1s) ([raw file](https://github.com/gramineproject/gramine/blob/b8cda8d6f242b8773a5f26281a13937d372f6366/Documentation/quickstart.rst#L15)):*
> ```reStructuredText
> - the FSGSBASE feature of recent processors must be enabled in the Linux kernel;
> - Linux kernel version at least 5.11 (with SGX driver enabled);
> - Intel SGX SDK/PSW and (optionally) Intel DCAP must be installed.
> ```

By the way, I would remove `Intel SGX SDK` from our descriptions. It is absolutely not needed for Gramine, and it may give an impression that Gramine builds on top of SDK.

The only things Gramine needs are some packages from Intel SGX PSW project and Intel DCAP (see the list here: https://github.com/gramineproject/gramine/blob/master/packaging/docker/Dockerfile).

Not blocking as this is orthogonal.
___
*[`Documentation/quickstart.rst` line 131 at r1](https://reviewable.io/reviews/gramineproject/gramine/1136#-NMDofWQ6m0hDbRRyIys:-NMDofWQ6m0hDbRRyIyt:b1h9ase) ([raw file](https://github.com/gramineproject/gramine/blob/b8cda8d6f242b8773a5f26281a13937d372f6366/Documentation/quickstart.rst#L131)):*
> ```reStructuredText
> found in a separate repository https://github.com/gramineproject/examples.
> 
> Please note that these sample applications are tested on Ubuntu 18.04 and 20.04.
> ```

I would just change to `Ubuntu`? We test on 22.04 for example too.
___
*[`Documentation/quickstart.rst` line 133 at r1](https://reviewable.io/reviews/gramineproject/gramine/1136#-NMDoWVY4t2qr2FDXJzU:-NMDoWVY4t2qr2FDXJzV:b-2gt9ey) ([raw file](https://github.com/gramineproject/gramine/blob/b8cda8d6f242b8773a5f26281a13937d372f6366/Documentation/quickstart.rst#L133)):*
> ```reStructuredText
> Please note that these sample applications are tested on Ubuntu 18.04 and 20.04.
> Most of these applications are also known to run correctly on
> Fedora/RHEL/CentOS, but with caveats. One caveat is that Makefiles should be
> ```

Maybe change the list of distros here too? `RHEL/AlmaLinux/RockyLinux`

**Review by woju — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1136)* status: 1 of 2 files reviewed, 4 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @dimakuv)

*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/1136#-NMDo8r09St3XvBsnnbS:-NMEIw7-7dnf3oMAQILH:b-3cacus) (no related file):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Where do we mention that `gramine-oot` and `gramine-dcap` are removed from the list of packages? Only in Release Notes? Not somewhere in the Docs?
</blockquote></details>

Yes, I think only in release notes. Docs describe current state.

___
___
*[`Documentation/quickstart.rst` line 15 at r1](https://reviewable.io/reviews/gramineproject/gramine/1136#-NMDn6261K97h-nKw_xl:-NMEO5VTAnzDWNO7Tqj2:bojjhi3) ([raw file](https://github.com/gramineproject/gramine/blob/b8cda8d6f242b8773a5f26281a13937d372f6366/Documentation/quickstart.rst#L15)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

By the way, I would remove `Intel SGX SDK` from our descriptions. It is absolutely not needed for Gramine, and it may give an impression that Gramine builds on top of SDK.

The only things Gramine needs are some packages from Intel SGX PSW project and Intel DCAP (see the list here: https://github.com/gramineproject/gramine/blob/master/packaging/docker/Dockerfile).

Not blocking as this is orthogonal.
</blockquote></details>

No, this should be corrected now, or won't be until next release. I've also removed mention of FSGSBASE patchset, because it was merged before upstream driver (point 2). Please check.

I'm happy to correct any other thing in one of those two documents, and later we'll just copy-paste to release notes.
___
*[`Documentation/quickstart.rst` line 131 at r1](https://reviewable.io/reviews/gramineproject/gramine/1136#-NMDofWQ6m0hDbRRyIys:-NMESWm51rSbmz2X1-e0:b-896fix) ([raw file](https://github.com/gramineproject/gramine/blob/b8cda8d6f242b8773a5f26281a13937d372f6366/Documentation/quickstart.rst#L131)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

I would just change to `Ubuntu`? We test on 22.04 for example too.
</blockquote></details>

Done.
___
*[`Documentation/quickstart.rst` line 133 at r1](https://reviewable.io/reviews/gramineproject/gramine/1136#-NMDoWVY4t2qr2FDXJzU:-NMESXoJC5B9BhPMpUZ-:b-896fix) ([raw file](https://github.com/gramineproject/gramine/blob/b8cda8d6f242b8773a5f26281a13937d372f6366/Documentation/quickstart.rst#L133)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Maybe change the list of distros here too? `RHEL/AlmaLinux/RockyLinux`
</blockquote></details>

Done.

**Review by dimakuv — DISMISSED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1136)* status: all files reviewed, all discussions resolved, not enough approvals from maintainers (1 more required)

___
*[`Documentation/quickstart.rst` line 15 at r1](https://reviewable.io/reviews/gramineproject/gramine/1136#-NMDn6261K97h-nKw_xl:-NMEbRXd6hFxnXAfBy7l:b-c1sv8z) ([raw file](https://github.com/gramineproject/gramine/blob/b8cda8d6f242b8773a5f26281a13937d372f6366/Documentation/quickstart.rst#L15)):*
<details><summary><i>Previously, woju (Wojtek Porczyk) wrote…</i></summary><blockquote>

No, this should be corrected now, or won't be until next release. I've also removed mention of FSGSBASE patchset, because it was merged before upstream driver (point 2). Please check.

I'm happy to correct any other thing in one of those two documents, and later we'll just copy-paste to release notes.
</blockquote></details>

Looks good to me now.

**Review by mkow — CHANGES_REQUESTED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1136)* status: all files reviewed, 1 unresolved discussion (waiting on @woju)

*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/1136#-NMnDN7M0F3ip8r-aiz2:-NMnDN7M0F3ip8r-aiz3:bvensou) (no related file):*
Blocking until @jinengandhi-intel verifies and approves these instructions.

___

**Review by jinengandhi-intel — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1136)* status: all files reviewed, 1 unresolved discussion (waiting on @mkow)

*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/1136#-NMnDN7M0F3ip8r-aiz2:-NMnUhBq5mezbrPQwHHP:bvqonx6) (no related file):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

Blocking until @jinengandhi-intel verifies and approves these instructions.
</blockquote></details>

@mkow I thought the understanding was to verify the instructions after Woju generates and uploads the `unstable-` packages from the Gramine v1.4 code freeze commit.

___

**Review by mkow — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1136)* status: all files reviewed, 1 unresolved discussion (waiting on @jinengandhi-intel)

*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/1136#-NMnDN7M0F3ip8r-aiz2:-NMnb5sr-QPIuVOp1rVy:bie762i) (no related file):*
<details><summary><i>Previously, jinengandhi-intel wrote…</i></summary><blockquote>

@mkow I thought the understanding was to verify the instructions after Woju generates and uploads the `unstable-` packages from the Gramine v1.4 code freeze commit.
</blockquote></details>

Yes, that's right. I blocked this PR to ensure that we won't accidentally merge it before that happens.

___

**Review by dimakuv — CHANGES_REQUESTED** [review]
> Reviewed 4 of 4 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1136)* status: all files reviewed, 4 unresolved discussions, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @jinengandhi-intel and @woju)

___
*[`Documentation/devel/building.rst` line 63 at r3](https://reviewable.io/reviews/gramineproject/gramine/1136#-NN3RJ-P8nm7gSyYGpI3:-NN3RJ-Q5AjSNK8xQgZs:b-gwsrpt) ([raw file](https://github.com/gramineproject/gramine/blob/0b7b7c4ea1490d9b7998ac707494f473a2e7d90b/Documentation/devel/building.rst#L63)):*
> ```reStructuredText
> infrastructure to be installed on the system. We require Linux kernel with SGX
> driver built in (``CONFIG_X86_SGX=y``, which is the case for most of available
> distribution kernels), which is available since version 5.11 (and also as a |~|
> ```

`as a` -> `as`
___
*[`Documentation/devel/building.rst` line 422 at r3](https://reviewable.io/reviews/gramineproject/gramine/1136#-NN3S3MFBhvJXLtBaZ9q:-NN3S3MG2xObLUzg7gvW:b-1z50pl) ([raw file](https://github.com/gramineproject/gramine/blob/0b7b7c4ea1490d9b7998ac707494f473a2e7d90b/Documentation/devel/building.rst#L422)):*
> ```reStructuredText
> 
> If you have an older CPU without :term:`FLC` support, you need to download and
> install the the following out-of-tree (OOT) Intel SGX driver:
> ```

`the the` -> `the`
___
*[`Documentation/devel/packaging.rst` line 12 at r3](https://reviewable.io/reviews/gramineproject/gramine/1136#-NN3SKWnEvXGBxmNDk8A:-NN3SKWnEvXGBxmNDk8B:bdgs5rf) ([raw file](https://github.com/gramineproject/gramine/blob/0b7b7c4ea1490d9b7998ac707494f473a2e7d90b/Documentation/devel/packaging.rst#L12)):*
> ```reStructuredText
> - Ubuntu (22.04LTS, 20.04LTS, 18.04LTS).
> - RHEL-8-like distributions (like AlmaLinux 8, Rocky Linux 8, ...).
> - experimentally, RHEL-9-like distros (packages are not fully validated)
> ```

Put a dot at the end

**Review by woju — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1136)* status: all files reviewed, 4 unresolved discussions, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @dimakuv and @jinengandhi-intel)

___
*[`Documentation/devel/building.rst` line 63 at r3](https://reviewable.io/reviews/gramineproject/gramine/1136#-NN3RJ-P8nm7gSyYGpI3:-NN3Sbk1Cm0xPJE71lzY:b-896fix) ([raw file](https://github.com/gramineproject/gramine/blob/0b7b7c4ea1490d9b7998ac707494f473a2e7d90b/Documentation/devel/building.rst#L63)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

`as a` -> `as`
</blockquote></details>

Done.
___
*[`Documentation/devel/building.rst` line 422 at r3](https://reviewable.io/reviews/gramineproject/gramine/1136#-NN3S3MFBhvJXLtBaZ9q:-NN3Sfhv61Lt2Kz_DmYS:b-896fix) ([raw file](https://github.com/gramineproject/gramine/blob/0b7b7c4ea1490d9b7998ac707494f473a2e7d90b/Documentation/devel/building.rst#L422)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

`the the` -> `the`
</blockquote></details>

Done.
___
*[`Documentation/devel/packaging.rst` line 12 at r3](https://reviewable.io/reviews/gramineproject/gramine/1136#-NN3SKWnEvXGBxmNDk8A:-NN3SmmYDgGieXNOJvM2:b-896fix) ([raw file](https://github.com/gramineproject/gramine/blob/0b7b7c4ea1490d9b7998ac707494f473a2e7d90b/Documentation/devel/packaging.rst#L12)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Put a dot at the end
</blockquote></details>

Done.

**Review by dimakuv — APPROVED** [review]
> Reviewed 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1136)* status: all files reviewed, 1 unresolved discussion (waiting on @jinengandhi-intel)

**Review by dimakuv — APPROVED** [review]
> Reviewed 2 of 2 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1136)* status: all files reviewed, 2 unresolved discussions (waiting on @jinengandhi-intel and @woju)

___
*[`Documentation/quickstart.rst` line 62 at r6](https://reviewable.io/reviews/gramineproject/gramine/1136#-NN6ggnx24fb1FrLdpx5:-NN6ggnx24fb1FrLdpx6:boq6hv6) ([raw file](https://github.com/gramineproject/gramine/blob/dbb53d27570724ad2117556265e73a20262f5fcb/Documentation/quickstart.rst#L62)):*
> ```reStructuredText
> 
> (like AlmaLinux, Rocky Linux, ...)
> ```

Accidentally have two empty lines, but I won't block

**Comment by woju** [comment]
Jenkins, test pkg-rpm-almalinux9 please

**Review by woju — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1136)* status: all files reviewed, 2 unresolved discussions (waiting on @dimakuv and @jinengandhi-intel)

___
*[`Documentation/quickstart.rst` line 62 at r6](https://reviewable.io/reviews/gramineproject/gramine/1136#-NN6ggnx24fb1FrLdpx5:-NN6w7SF7EiiBA3lU_Ws:bacarjz) ([raw file](https://github.com/gramineproject/gramine/blob/dbb53d27570724ad2117556265e73a20262f5fcb/Documentation/quickstart.rst#L62)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Accidentally have two empty lines, but I won't block
</blockquote></details>

Will fix during rebase.

**Review by woju — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1136)* status: 4 of 5 files reviewed, 2 unresolved discussions, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @dimakuv and @jinengandhi-intel)

___
*[`Documentation/quickstart.rst` line 62 at r6](https://reviewable.io/reviews/gramineproject/gramine/1136#-NN6ggnx24fb1FrLdpx5:-NN6wOOv1-UMGsZzj4jO:bvxsol9) ([raw file](https://github.com/gramineproject/gramine/blob/dbb53d27570724ad2117556265e73a20262f5fcb/Documentation/quickstart.rst#L62)):*
<details><summary><i>Previously, woju (Wojtek Porczyk) wrote…</i></summary><blockquote>

Will fix during rebase.
</blockquote></details>

Actually fixed right now, because there was an error in second point (dnf install epel-release).

**Review by dimakuv — DISMISSED** [review]
> Reviewed 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1136)* status: all files reviewed, 1 unresolved discussion (waiting on @jinengandhi-intel)

**Review by jinengandhi-intel — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1136)* status: all files reviewed, 1 unresolved discussion (waiting on @mkow)

*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/1136#-NMnDN7M0F3ip8r-aiz2:-NN7_0gxAsqYMp7XZW9l:bb6ywmx) (no related file):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

Yes, that's right. I blocked this PR to ensure that we won't accidentally merge it before that happens.
</blockquote></details>

LGTM

___

**Review by mkow — CHANGES_REQUESTED** [review]
> Reviewed 1 of 4 files at r3, 2 of 2 files at r5, 1 of 2 files at r6, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1136)* status: all files reviewed, 2 unresolved discussions (waiting on @woju)

___
*[`Documentation/cloud-deployment.rst` line 27 at r7](https://reviewable.io/reviews/gramineproject/gramine/1136#-NN7dAES4jDkirio8Ah2:-NN7dAES4jDkirio8Ah3:bj0qwf0) ([raw file](https://github.com/gramineproject/gramine/blob/a0b248f73df1d6016a9b657d4dbcbf6eab58f419/Documentation/cloud-deployment.rst#L27)):*
> ```reStructuredText
> ^^^^^^^^^^^^^^^
> 
> On Ubuntu 20.04LTS and 18.04LTS::
> ```

Why do we even mention this? There is no Ubuntu 18.04 non-LTS?



_Suggestion:_
```reStructuredText
Ubuntu 20.04 LTS and 18.04 LTS
```
___
*[`Documentation/devel/packaging.rst` line 10 at r7](https://reviewable.io/reviews/gramineproject/gramine/1136#-NN7fgUn10wLVqJErkPr:-NN7fgUn10wLVqJErkPs:b-7ropln) ([raw file](https://github.com/gramineproject/gramine/blob/a0b248f73df1d6016a9b657d4dbcbf6eab58f419/Documentation/devel/packaging.rst#L10)):*
> ```reStructuredText
> Currently officially supported distributions:
> 
> - Ubuntu (22.04LTS, 20.04LTS, 18.04LTS);
> ```

ditto

**Review by woju — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1136)* status: all files reviewed, 2 unresolved discussions (waiting on @mkow)

___
*[`Documentation/cloud-deployment.rst` line 27 at r7](https://reviewable.io/reviews/gramineproject/gramine/1136#-NN7dAES4jDkirio8Ah2:-NN7g1VY7cAY6BdDwbvS:b-pv4jir) ([raw file](https://github.com/gramineproject/gramine/blob/a0b248f73df1d6016a9b657d4dbcbf6eab58f419/Documentation/cloud-deployment.rst#L27)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

Why do we even mention this? There is no Ubuntu 18.04 non-LTS?
</blockquote></details>

Yes, there's no 18.04 non-LTS, but "LTS" is part of version and everyone who makes a list of ubuntu version writes that:

https://wiki.ubuntu.com/#Releases
https://wiki.ubuntu.com/Releases
https://en.wikipedia.org/wiki/Ubuntu_version_history

And for some time they were actually writing it together without space like here. I can also write with space, but it looks like too many LTS. See for yourself.
___
*[`Documentation/devel/packaging.rst` line 10 at r7](https://reviewable.io/reviews/gramineproject/gramine/1136#-NN7fgUn10wLVqJErkPr:-NN7iQiF92nw2rn5f8Wa:b-7ropln) ([raw file](https://github.com/gramineproject/gramine/blob/a0b248f73df1d6016a9b657d4dbcbf6eab58f419/Documentation/devel/packaging.rst#L10)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

ditto

**Review by mkow — APPROVED** [review]
> Reviewed 3 of 3 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1136)* status: all files reviewed, all discussions resolved, not enough approvals from different teams (1 more required, approved so far: ITL)

___
*[`Documentation/cloud-deployment.rst` line 27 at r7](https://reviewable.io/reviews/gramineproject/gramine/1136#-NN7dAES4jDkirio8Ah2:-NN7lZlV7paGXdjsQleS:bodo2lx) ([raw file](https://github.com/gramineproject/gramine/blob/a0b248f73df1d6016a9b657d4dbcbf6eab58f419/Documentation/cloud-deployment.rst#L27)):*
<details><summary><i>Previously, woju (Wojtek Porczyk) wrote…</i></summary><blockquote>

Yes, there's no 18.04 non-LTS, but "LTS" is part of version and everyone who makes a list of ubuntu version writes that:

https://wiki.ubuntu.com/#Releases
https://wiki.ubuntu.com/Releases
https://en.wikipedia.org/wiki/Ubuntu_version_history

And for some time they were actually writing it together without space like here. I can also write with space, but it looks like too many LTS. See for yourself.
</blockquote></details>

Looks good IMO.

**Review by dimakuv — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1136)* status: :shipit: complete! all files reviewed, all discussions resolved

**Pull request merged**
